automation_tools/__init__.py

```diff
@@ -2139,16 +2082,49 @@ def product_install(distribution, create_vm=False, certificate_url=None,
         test_in_stage = (test_in_stage.lower() == 'true')
 
     install_tasks = {
-        'satellite6-beta': cdn_install,
-        'satellite6-cdn': cdn_install,
-        'satellite6-downstream': downstream_install,
-        'satellite6-repofile': repofile_install,
-        'satellite6-activationkey': ak_install,
-        'satellite6-iso': iso_install,
-        'satellite6-upstream': upstream_install,
-        'satellite6-koji': upstream_install,
+        'beta': cdn_install,
+        'cdn': cdn_install,
+        'downstream': downstream_install,
+        'repofile': repofile_install,
+        'activationkey': ak_install,
+        'iso': iso_install,
+        'upstream': upstream_install,
+        'koji': upstream_install,
     }
-    distribution = distribution.lower()
+
+    # install task params population
+    task_params = {
+        'run_katello_installer': False,
+        'admin_password': os.environ.get('ADMIN_PASSWORD', 'changeme'),
+    }
+    if distribution in ('upstream', 'koji'):
+        task_params.update({'distribution': distribution})
+    else:
+        task_params.update({'sat_version': sat_version})
+
+    if distribution == 'repofile':
+        url = os.environ.get('REPO_FILE_URL')
+        if url:
+            task_params.update({'url': url})
+        else:
+            print('The REPO_FILE_URL environment variable should be defined')
+            sys.exit(1)
+    elif distribution == 'downstream':
+        url = os.environ.get('BASE_URL')
+        if url:
+            task_params.update({'url': url})
+        else:
+            print('The BASE_URL environment variable should be defined')
+            sys.exit(1)
+    elif distribution == 'iso':
+        check_gpg_signatures = os.environ.get('CHECK_GPG_SIGNATURES', '').lower() == 'true'
+        task_params.update({'check_gpg_signatures': check_gpg_signatures})
+        url = os.environ.get('ISO_URL') or os.environ.get('BASE_URL')
```

Comment:
Why do we have `or` here? In the previous version it works only with `ISO_URL`. But, frankly, I have not investigated full logic here.

Reply:
just reading it, that allows for it to use either field name. in the case an ISO_URL field isn't available/filled, then we can use the BASE_URL field.

Reply:
True.